### PR TITLE
[OpenAI.Inference] Adding models to represent streaming requests for completions and chat completions.

### PR DESCRIPTION
**Pull request by glecaros**

### CODE REVIEW

File: specification/cognitiveservices/OpenAI.Inference/client.tsp [hunk 1]
```diff
@@ -12,3 +12,12 @@ using Azure.ClientGenerator.Core;
 @@include(Azure.OpenAI.ImageGenerationOptions);
 @@include(Azure.OpenAI.ImageLocation);
 @@include(Azure.OpenAI.ImageGenerations);
+
```
Comment (trrwilson) [inline specification/cognitiveservices/OpenAI.Inference/client.tsp:15]: explanatory comment will be very helpful for future readers

File: specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp [hunk 2]
```diff
@@ -176,75 +201,113 @@ model ChatCompletionsOptions {
...
-The representation of a single prompt completion as part of an overall chat completions request.
-Generally, `n` choices are generated per provided prompt with a default value of 1.
-Token limits and other settings may limit the number of choices generated.
-""")
-model ChatChoice {
-  @doc("The chat message for a given chat completions prompt.")
-  @projectedName("json", "message")
-  message?: ChatMessage;
-
+@doc("Model representing the common base between chat choice types.")
```
Comment (trrwilson) [inline specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp:204]: May be worth commenting (or adding to doc) that it's not meant to be a user-facing type and should just be consolidated representation for spread (...) operators.

File: specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp [hunk 3]
```diff
@@ -176,75 +201,113 @@ model ChatCompletionsOptions {
...
+  Content filtering results for zero or more prompts in the request. In a streaming request,
+  results for different prompts may arrive at different times or in different orders.
+  """)
+  @added(ServiceApiVersions.v2023_06_01_Preview)
+  @projectedName("json", "prompt_annotations")
+  promptFilterResults?: PromptFilterResult[];
+}
+
+@doc("Representation of the response data from a streaming chat completions request.")
+model ChatCompletionsChunk {
```
Comment (trrwilson) [inline specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp:277]: Should we consider still using the word "Streaming" somewhere, like `ChatCompletionsStreamingChunk`? Actual, not rhetorical question :)

Comment (glecaros) [reply specification/cognitiveservices/OpenAI.Inference/models/chat.completions.tsp:277]: I'm not sure about that... we stream chunks but chunks are not the streaming. Also, I essentially lifted the name from the `object` property OAI returns.

**Pull request closed without merging**